templates/Makefile.tpl

```diff
@@ -0,0 +1,53 @@
+# Current Operator version
+VERSION ?= {{ .Version }}
+# Default bundle image tag
+BUNDLE_IMG ?= aws-{{.ServiceIDClean}}-controller-bundle:$(VERSION)
+# Options for 'bundle-build'
+ifneq ($(origin CHANNELS), undefined)
+BUNDLE_CHANNELS := --channels=$(CHANNELS)
+endif
+ifneq ($(origin DEFAULT_CHANNEL), undefined)
+BUNDLE_DEFAULT_CHANNEL := --default-channel=$(DEFAULT_CHANNEL)
+endif
+BUNDLE_METADATA_OPTS ?= $(BUNDLE_CHANNELS) $(BUNDLE_DEFAULT_CHANNEL)
+
+# Image URL to use all building/pushing image targets
+IMG ?= {{.Annotations.ContainerImage}}:{{.ServiceIDClean}}-v{{.Version}}
+
+# Get the currently used golang install path (in GOPATH/bin, unless GOBIN is set)
+ifeq (,$(shell go env GOBIN))
+GOBIN=$(shell go env GOPATH)/bin
+else
+GOBIN=$(shell go env GOBIN)
+endif
+
+# Download kustomize locally if necessary
+KUSTOMIZE = $(shell pwd)/bin/kustomize
+kustomize:
+	$(call go-get-tool,$(KUSTOMIZE),sigs.k8s.io/kustomize/kustomize/v3@v3.8.7)
+
+# go-get-tool will 'go get' any package $2 and install it to $1.
+PROJECT_DIR := $(shell dirname $(abspath $(lastword $(MAKEFILE_LIST))))
+define go-get-tool
+@[ -f $(1) ] || { \
+set -e ;\
+TMP_DIR=$$(mktemp -d) ;\
+cd $$TMP_DIR ;\
+go mod init tmp ;\
+echo "Downloading $(2)" ;\
+GOBIN=$(PROJECT_DIR)/bin go get $(2) ;\
+rm -rf $$TMP_DIR ;\
+}
+endef
+
+# Generate bundle manifests and metadata, then validate generated files.
+.PHONY: bundle
+bundle: kustomize
+	cd config/controller && $(KUSTOMIZE) edit set image controller=$(IMG)
+	$(KUSTOMIZE) build config/manifests | operator-sdk generate bundle -q --overwrite --version $(VERSION) $(BUNDLE_METADATA_OPTS) --package ack-{{ .ServiceIDClean }}-controller
+	operator-sdk bundle validate ./bundle
+
+# Build the bundle image.
+.PHONY: bundle-build
+bundle-build:
+	docker build -f bundle.Dockerfile -t $(BUNDLE_IMG) .
```

Comment:
Is the above Makefile necessary for OLM to work? Looking at the above, I'd prefer to modify the `scripts/build-controller-release.sh` script in the main community source repository to optionally call `operator-sdk generate bundle` instead of generating a templatized Makefile like this...

Reply:
Sure thing, it's not required - just a nicety. I'll have to get more familiar with the script but given that it's a separate repo, I can work on a separate PR there. I'll drop the Makefile and submit another commit shortly.

Reply:
The Makefile template has been removed and is no longer part of the rendering logic

Reply:
I've started playing with https://github.com/aws-controllers-k8s/community/blob/main/scripts/build-controller-release.sh and am wondering if we should recreate all targets there or put them in a separate script. I'm also thinking the calls out to build the image using docker can probably be left to "Documentation".

I've put together the `ack-generate olm` into that script and that makes sense to me, but doing the bundle generation (i.e. targets that lay down bundle/ and bundle.Dockerfile` requires OperatorSDK. If it's optional, we don't want to require OperatorSDK be installed for folks who are not interested. Don't want to cause extra work for those folks.

As a side note, should I go ahead and start an issue in community so we can discuss this separately?

Reply:
> I've started playing with https://github.com/aws-controllers-k8s/community/blob/main/scripts/build-controller-release.sh and am wondering if we should recreate all targets there or put them in a separate script. I'm also thinking the calls out to build the image using docker can probably be left to "Documentation".
> 
> I've put together the `ack-generate olm` into that script and that makes sense to me, but doing the bundle generation (i.e. targets that lay down bundle/ and bundle.Dockerfile` requires OperatorSDK. If it's optional, we don't want to require OperatorSDK be installed for folks who are not interested. Don't want to cause extra work for those folks.
> 
> As a side note, should I go ahead and start an issue in community so we can discuss this separately?

@komish yes, please do open a new issue for this topic. sorry about delay in reviewing this. Will try to get a full review done today!

Reply:
I'll get an issue created and CC you in the coming week. Good to resolve here when you are.